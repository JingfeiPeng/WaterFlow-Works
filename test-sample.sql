-- Query for finding all the prerequisite courses 
-- We can figure out whether this is an All_of group or One_of group 
--   based on whether the rows belong to the same coursegroupid 
SELECT coursecode, coursegroupid
FROM courseGroupMember
WHERE courseGroupID IN (
    SELECT groupid
    FROM courseGroup
    WHERE groupID IN (
        SELECT prereqcoursegroupid
        FROM prerequisite
        WHERE courseCode = UPPER('cs 488')
    )
);

-- Query to get the basic information about a course like title, description, etc
SELECT courseCode,title, courseTypes, description, subjectTitle 
FROM course 
WHERE UPPER(courseCode) = UPPER('cs 488');