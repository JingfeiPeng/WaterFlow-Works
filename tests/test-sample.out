 -> Result for the query to search for the prerequistes of a course
  coursecode | coursegroupid
------------+---------------
 CS 350     |            73
 CS 370     |            74
 CS 371     |            74
 CS 341     |            72
(4 rows)

 -> Results for the second query
 coursecode |               title               | coursetypes |  description                                                                                                                                                                                                                                                               | subjecttitle                    
------------+-----------------------------------+-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------
 CS 488     | Introduction to Computer Graphics | LAB,LEC,TST | Software and hardware for interactive computer graphics. Implementation of device drivers, 3-D transformations, clipping, perspective, and input routines. Data structures, hidden surface removal, colour shading techniques, and some additional topics will be covered. | CS
 (1 row)